Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -265,6 +288,106 @@ - (void)urlSessionTask:(NSURLSessionTask *)sessionTask setState:(NSURLSessionTas
     SENTRY_LOG_DEBUG(@"SentryNetworkTracker finished HTTP span for sessionTask");
 }
 
+- (void)captureFailedRequests:(NSURLSessionTask *)sessionTask
+{
+    // if request or response are null, we can't raise the event
+    if (sessionTask.currentRequest == nil || sessionTask.response == nil) {
+        return;
+    }
+    // some properties are only available if the response is of the NSHTTPURLResponse type
+    // bail if not
+    if (![sessionTask.response isKindOfClass:[NSHTTPURLResponse class]]) {
+        return;
+    }
+    NSHTTPURLResponse *myResponse = (NSHTTPURLResponse *)sessionTask.response;
+    NSURLRequest *myRequest = sessionTask.currentRequest;
+    NSNumber *responseStatusCode = @(myResponse.statusCode);
+
+    if (!self.isCaptureFailedRequestsEnabled || ![self containsStatusCode:myResponse.statusCode]) {
+        return;
+    }
+
+    if (![self isTargetMatch:myRequest.URL withTargets:SentrySDK.options.failedRequestTargets]) {
+        return;
+    }
+    
+// TODO: how would I conver the error to sentry's error? thsi could be useful in case something goes wrong
+//    sessionTask.error
```

Comment:
@brustolin @philipphofmann do you think this should be part of the event? how/where do I set this in the event?

Reply:
Maybe I could mimic `captureError` and add the exception as part of the exceptions list, what do you think?
Or do you think this should not be part of the event? I don't know what the `sessionTask.error` looks like, its always `nil` for me, not sure how to reproduce it.

Reply:
I dont know if I totally understand this, but SentryEvent has a constructor that you pass an NSError.

Reply:
`sessionTask.error` is mostly network related errors, and I do believe this may be useful.

Reply:
The question is if I should create a `SentryEvent` out of the `sessionTask.error`, or as I am currently doing by creating a synthetic exception only:

>     SentryException *sentryException = [[SentryException alloc] initWithValue:message
>                                                                          type:@"HTTPClientError"];

The other option is having a `SentryEvent` with both exceptions if `sessionTask.error` isn't nullable.
Since I can't reproduce this `sessionTask.error`, I can't really tell if its useful.
If you tell its a networking error, its not useful because this integration only creates events out of failed requests, so you technically didn't have a network error, the API returned an error.

Reply:
If it's either one or the other, I should ignore `error` completely because I can only capture errors within the range of status code.

Reply:
I believe @philipphofmann is right, and I agree with @marandaneto that this feature should only address server side errors. 

If the app crashes because there is no error handling for an app without network we will catch this crash anyway.